kernel/src/fs/fat12.c

```
@@ -657,6 +664,190 @@ fs_node_t *fat12_create(const char *name)
+        free(fctx.result);
+        return -1;
+    }
+
+    uint32_t cluster = fat12_allocate_cluster();
+    if (cluster == 0)
+        return -1;
+
+    /* Initialize the cluster with . and .. entries */
+    uint32_t bytes_per_cluster = fat12.bpb.sectors_per_cluster * fat12.bpb.bytes_per_sector;
+    uint8_t *cluster_buf = malloc(bytes_per_cluster);
+    memset(cluster_buf, 0, bytes_per_cluster);
```

Cppcheck (2.21.0)

[warning] If memory allocation fails, then there is a possible null pointer dereference

(nullPointerOutOfMemory)